x-pack/winlogbeat/module/security/_meta/kibana/7/dashboard/71f720f0-ff18-11e9-8405-516218e3d268.json

```diff
@@ -0,0 +1,4399 @@
+          "searchSourceJSON": {
+            "filter": [
+              {
+                "$state": {
+                  "store": "appState"
+                },
+                "meta": {
+                  "alias": null,
+                  "disabled": false,
+                  "indexRefName": "kibanaSavedObjectMeta.searchSourceJSON.filter[0].meta.index",
+                  "key": "event.code",
+                  "negate": false,
+                  "params": {
+                    "query": "4720"
+                  },
+                  "type": "phrase",
+                  "value": "4720"
+                },
+                "query": {
+                  "match": {
+                    "event.code": {
+                      "query": "4720",
+                      "type": "phrase"
+                    }
+                  }
+                }
+              }
+            ],
+            "indexRefName": "kibanaSavedObjectMeta.searchSourceJSON.index",
+            "query": {
+              "language": "kuery",
+              "query": ""
+            }
+          }
+        },
+        "title": "Users Created - Table",
+        "uiStateJSON": {
+          "vis": {
+            "params": {
+              "sort": {
+                "columnIndex": null,
+                "direction": null
+              }
+            }
+          }
+        },
+        "version": 1,
+        "visState": {
+          "aggs": [
+            {
+              "enabled": true,
+              "id": "1",
+              "params": {},
+              "schema": "metric",
+              "type": "count"
+            },
+            {
+              "enabled": true,
+              "id": "2",
+              "params": {
+                "customLabel": "Created User",
+                "field": "winlog.event_data.TargetUserName",
+                "missingBucket": false,
+                "missingBucketLabel": "Missing",
+                "order": "desc",
+                "orderBy": "1",
+                "otherBucket": false,
+                "otherBucketLabel": "Other",
+                "size": 100
+              },
+              "schema": "bucket",
+              "type": "terms"
+            },
+            {
+              "enabled": true,
+              "id": "3",
+              "params": {
+                "customLabel": "Performed by",
+                "field": "winlog.event_data.SubjectUserName",
+                "missingBucket": true,
+                "missingBucketLabel": "Missing",
+                "order": "desc",
+                "orderBy": "1",
+                "otherBucket": false,
+                "otherBucketLabel": "Other",
+                "size": 5
+              },
+              "schema": "bucket",
+              "type": "terms"
+            },
+            {
+              "enabled": true,
+              "id": "4",
+              "params": {
+                "customLabel": "Performer LogonID",
+                "field": "winlog.logon.id",
```

Comment:
It appears a user from the community (Using a fresh 7.8.0 stack) noticed that winlog.logon.id does not exist in the winlogbeat-* index pattern from the setup which yields in an error message "Could not locate that index-pattern-field (id: winlog.logon.id)" when trying to edit/look at some of the new visualizations such as "Users Deleted - Table [Winlogbeat Security]"

Not sure if I follow but it looks like the index pattern is "generated" from the fields.yml and if that is true I can see why this field does not get added to the winlogbeat-* index pattern.

The fields.common.yml is missing winlog.logon.id so maybe that is why the default index pattern that gets generated may not have this field?

Slack thread here: https://elasticstack.slack.com/archives/CNEDGGJQ3/p1595052755138500

Maybe this is a case of Dashboards/Visualizations using a custom index pattern that isn't fully compliant with the default winlogbeat-* index pattern created from the setup?

If users don't get all of the event logs that would contain these fields then the index pattern would never contain this field, so maybe it is worth adding into the fields.common.yml? I just wanted to make a note of this analysis.

Reply:
Hi @nicpenning,
I don't know exactly how the winlogbeat index pattern is generated, but probably I'll need to add the field winlog.logon.id in the security module specific fields.yml file (beats\x-pack\winlogbeat\module\security\_meta\fields.yml)
@andrewkroh, what do you think?  Should I add this field here or in any other place?
In addition. Which is the criteria in order to add a field in the yaml? The winlog.logon.id is not the only field I've added; I can add the necessary fields.
Thank you!
Regards